grasp_generation/utils/isaac_validator.py

```diff
@@ -884,6 +884,33 @@ def save_images(self, folder: str, overwrite: bool = False) -> None:
         # Avoid segfault if run multiple times by destroying camera sensors
         self._destroy_cameras(self.envs[0])
 
+    def save_images_lightweight(
+        self,
+        folder: str,
+        obj_scale: float,
+        overwrite: bool = False,
+        generate_seg: bool = False,
+        generate_depth: bool = False,
+    ) -> None:
+        assert len(self.envs) == 1
+        self._setup_cameras(self.envs[0], radius=2 * obj_scale)
+
+        gym.step_graphics(self.sim)
+        gym.render_all_camera_sensors(self.sim)
+        path = self._setup_save_dir(folder, overwrite)
+
+        for ii, camera_handle in enumerate(self.camera_handles):
+            self._save_single_image_lightweight(
+                path,
+                ii,
+                camera_handle,
+                generate_seg=generate_seg,
+                generate_depth=generate_depth,
+            )
+
+        # Avoid segfault if run multiple times by destroying camera sensors
```

Comment:
Nice!

Reply:
this was actually already there lol, it's just showing up in the diff because I made a new function that was mostly copy-paste of the old one

Reply:
also, it doesn't avoid segfault, lmao, but the old code didn't either, so...